### DISCUSSION TITLE
Enhancement - Organize files by Year - Month

### DISCUSSION THREAD

**Discussion by Eagle248** [djdd87/SynoAI#163] [Ideas]
Possible to Organize files by Year - Month. By doing this way, there will be less number of files in 1 folder. As of right now after few months thousands of file will be under 1 folder "Camera1" which becomes hard to navigate.
Thanks.
Example:
/Captures/Camera1/2022/2022-10

**Comment by djdd87** [comment]
Yeah, I can look into this. Something like Year - Month - Day would be better in my opinion?

**Comment by Eagle248** [comment]
Ideally strcture should be what i think camera1/2022/2022-10/file1.jpg.

folders by date as well might be useful to some users as well. May be see
if possible to provide folder structure as a variable, so user can
customize as per there own.

And delete old files automatically older than X days.


Look at this project,  if it is helpful to you.

Environment variable:

folder_structure: This specifies the folder structure to use in your
download destination directory. If this variable is not set, it will set
{:%Y/%m/%d} as the default. Use none to download to a flat file structure.



https://github.com/boredazfcuk/docker-icloudpd

**Reply by djdd87** [reply]
> And delete old files automatically older than X days.

This functionality already exists.

**Comment by Eagle248** [comment]
I might have missed this in the read me file on deleting files after xx
days.

**Reply by Eagle248** [reply]
Hi, did you got the chance to work on adding folders by year/ month/ date?

**Reply by djdd87** [reply]
Not yet, I've had an incredibly busy year work-wise so far. I'll try and remember to do this as it shouldn't take too long.

**Reply by Eagle248** [reply]
Are you planning to follow the logic like
Camera1 (main folder)
 - 2023 (sub folder)
 -- 2023-04 (Sub folder)
--- 2023-04-13 (sub folder)

**Reply by djdd87** [reply]
I was probably thinking more like:
- year
  - month
    - day

E.g.

2023
  04 
     13

Although maybe your suggestion is better?

**Reply by Eagle248** [reply]
The reason i said yyyy, yyyy-mm, yyyy-mm-dd because if someone copy the
folder then they do not need to keep track of the only date or only month
folder.

**Comment by Eagle248** [comment]
Thank you.
